src/hooks/chain/useCoinChain.ts

```diff
@@ -0,0 +1,107 @@
+import { QueryFunctionContext } from '@tanstack/react-query'
+
+import { CoinName, coinNameToTypeMap, EvmCoinType } from '@ensdomains/address-encoder'
+import { coinTypeToEvmChainId, isEvmCoinType } from '@ensdomains/address-encoder/utils'
+
+import { SupportedAddress, supportedAddresses } from '@app/constants/supportedAddresses'
+import { useQueryOptions } from '@app/hooks/useQueryOptions'
+import { CreateQueryKey, QueryConfig } from '@app/types'
+import { getIsCachedData } from '@app/utils/getIsCachedData'
+import { prepareQueryOptions } from '@app/utils/prepareQueryOptions'
+import { useQuery } from '@app/utils/query/useQuery'
+
+type UseCoinChainParameters = {
+  coinName?: string
+}
+
+type CoinBlockExplorer = {
+  id: number
+  name: string
+  nativeCurrency: {
+    name: string
+    symbol: string
+    decimals: number
+  }
+  blockExplorers: {
+    default: {
+      name: string
+      url: string
+      apiUrl?: string
+    }
+  }
+}
+
+type UseCoinChainReturnType = {
+  error: boolean
+  data: CoinBlockExplorer | null
+}
+
+type UseCoinChainConfig = QueryConfig<UseCoinChainReturnType, Error>
+
+type QueryKey<TParams extends UseCoinChainParameters> = CreateQueryKey<
+  TParams,
+  'getCoinChain',
+  'independent'
+>
+
+export const getCoinChainQueryFn = async <TParams extends UseCoinChainParameters>({
+  queryKey: [{ coinName }],
+}: QueryFunctionContext<QueryKey<TParams>>): Promise<UseCoinChainReturnType> => {
+  if (!coinName) throw new Error('name is required')
+
+  const lowerCoinName = coinName?.toLowerCase() || ''
+  const coinType = coinNameToTypeMap[lowerCoinName as CoinName]
+
+  if (supportedAddresses.includes(coinName?.toLowerCase() as SupportedAddress)) {
+    const supportedBlockExplorers = await import('../../constants/blockExplorers/supported.json')
+    const coinBlockExplorer = supportedBlockExplorers?.default.find(
+      (blockExplorer) =>
+        (isEvmCoinType(coinType) &&
+          blockExplorer?.id === coinTypeToEvmChainId(coinType as EvmCoinType)) ||
+        blockExplorer?.nativeCurrency?.symbol.toLowerCase() === lowerCoinName, // this is needed for certain coins such as eth, btc, sol, etc.
```

Comment:
*nativeCurrency.symbol* does not always match coinName. coinName is a symbol that we created to make it more user friendly to when working with multiple chains. So instead of seeing numbers like 60, 1093234, etc, we would just see eth, base, etc.

See comment in supported.json on how we should handle this.

Reply:
updated logic to use the coinName to get the blockexplorer for both supported and evm coins